python/lsst/ts/scheduler/utils/efd_utils.py

```diff
@@ -0,0 +1,215 @@
+# This file is part of ts_scheduler
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["get_efd_client", "get_mock_efd_client"]
+
+import pandas
+import unittest
+import warnings
+
+import numpy as np
+
+from typing import List
+
+try:
+    import lsst_efd_client
+
+    __with_lsst_efd_client__ = True
+except ImportError:
+    warnings.warn("Could not import lsst_efd_client. Telemetry stream will not work.")
+    __with_lsst_efd_client__ = False
+
+
+def get_efd_client(efd_name, mock=False):
```

Comment:
I see you have a mix of typed and non-typed code. I'd like to suggest to use typed code but perhaps this can be done in a future ticket?

Reply:
This is a tricky one... I'm honestly not sure how to represent this, because this method will either return a a `mock` or a class that may or may not be importable.. .

Reply:
well, I'll work around it for now and set it to `Any`.. hehehe